airflow/models/dag.py

```diff
@@ -3032,6 +3034,16 @@ def bulk_sync_to_db(
         )
         return cls.bulk_write_to_db(dags=dags, session=session)
 
+    def simplify_dataset_expression(self, dataset_expression) -> dict | None:
```

Comment:
This doesn’t need to be a method, a simple function will do. We can fix later, let’s merge this first to unblock frontend work.